File: CodeWars/7Kyu/bubble_sort.py
def bubble(lst):
    snapshots = []
    sorted_list = sorted(lst)
    
    if lst == sorted_list:
        return []
    
    for n in range(len(lst) - 1, 0, -1):
        swapped = False
        
        for i in range(n):
            if lst[i] > lst[i + 1]:
                lst[i], lst[i + 1] = lst[i + 1], lst[i]
                snapshots.append(lst[:])
                swapped = True
        
        if not swapped:
            break
            
    return snapshots

File: Ruby/bubble_sort.rb
def bubble_sort(array)
  length = array.length

  for i in 0...length
    swapped = false
    for j in 0...(length - i - 1)
      if array[j] > array[j + 1]
        array[j], array[j + 1] = array[j + 1], array[j]
        swapped = true
      end
    end
    if swapped == false
      break
    end
  end
  return array
end

p bubble_sort([4,3,78,2,0,2])

